libecho/src/RawReceiver.cc

```diff
@@ -0,0 +1,108 @@
+#include <cmath>
+#include "RawReceiver.h"
+
+#define SAMPLE_RATE 44100
+
+static double dft(const int16_t *buffer, int samples, double ratio) {
+    double re = 0;
+    double im = 0;
+    double angle = 0;
+    const double d_angle = 2.0 * M_PI * ratio;
+
+    for (int i = 0; i < samples; i++) {
+        double val = buffer[i];
+        re += val * std::cos(angle);
+        im += val * std::sin(angle);
+        angle += d_angle;
+    }
+
+    return std::sqrt(re * re + im * im) / samples;
+}
+
+RawReceiver::RawReceiver(int loFreq, int hiFreq, double magLimitBegin, double magLimitEnd, int winSize) {
```

Comment:
Initializer list ;)

Reply:
what?

Reply:
Use initializer list in constructor - instead of `this->loFreq = loFreq;` in constructor body, do something like this:
`RawReceiver::RawReceiver(int loFreq, int hiFreq, double magLimitBegin, double magLimitEnd, int winSize) : loFreq(loFreq) {`